### PR TITLE
Catch tags that do not exist

### LINKED ISSUE #4817: Uncaught exception when looking for a tag that doesn't exist
### Expected behavior

Exception is handled and the loop continues while ignoring the missing tag entries

### Actual behavior
Uncaught DoesNotExist Exception in the TagMapper

Can be fixed by adding a try catch to the foreach loop in TagMapper:158:

```php
			try {
				$result[] = $qr['imap_message_id'];
				$result[$qr['imap_message_id']][] = $this->getTag((int)$qr['tag_id']);
			} catch (DoesNotExistException $e) {
				continue;
			}
```

### PR DESCRIPTION
**Pull request by miaulalala**
Fixes #4817

### CODE REVIEW

File: lib/Db/TagMapper.php [hunk 1]
```diff
@@ -156,9 +156,13 @@ public function getAllTagsForMessages(array $messages): array {
 		}
 		$result = [];
 		foreach ($queryResult as $qr) {
-			$result[] = $qr['imap_message_id'];
-			$result[$qr['imap_message_id']][] = $this->getTag((int)$qr['tag_id']);
-		};
+			try {
+				$result[] = $qr['imap_message_id'];
```
Comment (st3iny) [inline lib/Db/TagMapper.php:160]: What exactly is the purpose of this line? IIRC, the returned result is expected to be a map of `message_id => tag[]` while this line is also creating an array of message ids.

Example output for `$result` (message with one tag):
```php
array (
  0 => '<20210308112210.Horde.djiH_5Ms4hawgi2WUoVHzTD@localhost>',
  '<20210308112210.Horde.djiH_5Ms4hawgi2WUoVHzTD@localhost>' => 
  array (
    0 => 
    OCA\Mail\Db\Tag::__set_state(array(
       'userId' => 'admin',
       'displayName' => 'Important',
       'imapLabel' => '$label1',
       'color' => '#FF0000',
       'isDefaultTag' => true,
       'id' => 1,
       '_updatedFields' => 
      array (
      ),
       '_fieldTypes' => 
      array (
        'id' => 'integer',
        'isDefaultTag' => 'boolean',
      ),
    )),
  ),
)
```

**Review by st3iny — APPROVED** [review]
Code looks good and works. However, I have one possibly unrelated question about the code.

File: lib/Db/TagMapper.php (hunk 1 above)
Comment (miaulalala) [reply lib/Db/TagMapper.php:160]: Yep, it was really not well written. will be removed with #4822

**Comment by miaulalala** [comment]
This is irrelevant as the method body is rewritten entirely in #4822

**Pull request closed without merging**